### ISSUE TITLE
Scheduled scans don't start for no obvious reason

### ISSUE THREAD

**Issue by ksamag** [Patrowl/PatrowlManager#175]
Hi. We're using Patrowl version 1.6.6 Community Edition and we have a couple of scheduled nmap scans.
<img width="1044" alt="Screen Shot 2020-10-31 at 5 57 37 PM" src="https://user-images.githubusercontent.com/66267604/97781740-a877af80-1ba2-11eb-9037-6f5937a53a13.png">
As you can see in the image, one of our scans has failed(as depicted below, no known reason for it):
![image](https://user-images.githubusercontent.com/66267604/97781759-c513e780-1ba2-11eb-8a38-9aede4b64293.png)
And after that, scans are just created and not started. Events in the scans which are not being done are as below:
![image](https://user-images.githubusercontent.com/66267604/97781784-e379e300-1ba2-11eb-8c2e-aa25b4b29435.png)

Which is strange. A fine-working scan should be assigned to an engine and then being started, but these scans are just created and not being started.

Any ideas why?

**Comment by ksamag** [comment]
They're stuck at `started` status and don't change to `running`

**Comment by ksamag** [comment]
I found the following message in celery logs:
```
(env3) root@patrowl-7b75d57f6f-gpxfd:/opt/patrowl-manager# cat var/log/celeryderr.scan.log
/opt/patrowl-manager/env3/lib/python3.7/site-packages/celery/platforms.py:801: RuntimeWarning: You're running the worker with superuser privileges: this is
absolutely not recommended!
Please specify a different user using the --uid option.
User information: uid=0 euid=0 gid=0 egid=0
  uid=uid, euid=euid, gid=gid, egid=egid,
[2020-11-07 12:33:22,309: INFO/MainProcess] Connected to amqp://rabbitmq:**@patrowl-rabbitmq:5672//
[2020-11-07 12:33:22,367: WARNING/MainProcess] /opt/patrowl-manager/env3/lib/python3.7/site-packages/celery/fixups/django.py:202: UserWarning: Using settings.DEBUG leads to a memory leak, never use this setting in production environments!
  warnings.warn('Using settings.DEBUG leads to a memory leak, never '
[2020-11-07 12:33:22,367: INFO/MainProcess] scan-node@patrowl-7b75d57f6f-gpxfd ready.
[2020-11-07 12:44:17,623: INFO/MainProcess] Received task: engines.tasks.startscan_task[2f6ac0b4-ce44-4937-aee5-c326c8ec69b5]  ETA:[2020-11-07 12:44:18.590655+00:00]
[2020-11-07 12:44:27,787: WARNING/ForkPoolWorker-32] scan status (in loop): FINISHED
[2020-11-07 12:44:32,963: INFO/ForkPoolWorker-32] Task engines.tasks.startscan_task[2f6ac0b4-ce44-4937-aee5-c326c8ec69b5] succeeded in 10.491162520833313s: True
[2020-11-07 12:45:19,041: INFO/MainProcess] Received task: engines.tasks.start_periodic_scan_task[874637d5-3210-430d-8d5b-6ac04090947f]
[2020-11-07 12:45:19,121: ERROR/ForkPoolWorker-32] Task engines.tasks.start_periodic_scan_task[874637d5-3210-430d-8d5b-6ac04090947f] raised unexpected: KeyError('assets')
Traceback (most recent call last):
  File "/opt/patrowl-manager/env3/lib/python3.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/opt/patrowl-manager/env3/lib/python3.7/site-packages/celery/app/trace.py", line 650, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/patrowl-manager/engines/tasks.py", line 569, in start_periodic_scan_task
    for asset in params['scan_params']['assets']:
KeyError: 'assets'
[2020-11-07 13:18:59,732: INFO/MainProcess] Received task: engines.tasks.startscan_task[f6684c4c-eb3e-42b0-bb2e-92cfa0571c91]  ETA:[2020-11-07 13:19:00.727608+00:00]
[2020-11-07 13:19:04,047: INFO/MainProcess] Received task: engines.tasks.startscan_task[91d0f157-80a6-4ca7-9f89-8ec744a3bff6]  ETA:[2020-11-07 13:19:05.044847+00:00]
[2020-11-07 13:19:07,979: WARNING/ForkPoolWorker-32] scan status (in loop): FINISHED
[2020-11-07 13:19:13,163: INFO/ForkPoolWorker-32] Task engines.tasks.startscan_task[f6684c4c-eb3e-42b0-bb2e-92cfa0571c91] succeeded in 10.357417287304997s: True
[2020-11-07 13:19:14,501: WARNING/ForkPoolWorker-1] scan status (in loop): FINISHED
[2020-11-07 13:19:19,698: INFO/ForkPoolWorker-1] Task engines.tasks.startscan_task[91d0f157-80a6-4ca7-9f89-8ec744a3bff6] succeeded in 10.495384318754077s: True
[2020-11-07 13:20:37,409: INFO/MainProcess] Received task: engines.tasks.start_periodic_scan_task[7be35ddf-4678-468c-8086-d458256effaa]
[2020-11-07 13:20:37,495: ERROR/ForkPoolWorker-32] Task engines.tasks.start_periodic_scan_task[7be35ddf-4678-468c-8086-d458256effaa] raised unexpected: KeyError('assets')
Traceback (most recent call last):
  File "/opt/patrowl-manager/env3/lib/python3.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/opt/patrowl-manager/env3/lib/python3.7/site-packages/celery/app/trace.py", line 650, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/patrowl-manager/engines/tasks.py", line 569, in start_periodic_scan_task
    for asset in params['scan_params']['assets']:
KeyError: 'assets'
```

**Comment by ksamag** [comment]
```
{
    "scan_params": {
        "engine_id": 3
    },
    "scan_definition_id": 27,
    "engine_name": "recognizer",
    "owner_id": 1,
    "engine_id": 3
}
```
`params` value

**Comment by sebastien-powl** [comment]
Fixed by #178 , merged into develop, will be fixed in next release on master branch.
